include/agent.h

```diff
@@ -0,0 +1,96 @@
+#ifndef _ABM_AGENT_H_
+#define _ABM_AGENT_H_
+
+#include <memory>
+#include <vector>
+
+#include "config.h"
+#include "graph.h"
+#include "agent.h"
+
+namespace abm {
+
+//! Agent class that has id, origin, destination, departure time, current node and compute path from current node to destination.
+class Agent {
+ public:
+  //! Construct an agent class
+  //! \param[in] graph Pass a reference of the graph that the agent does routing on
+  explicit Agent(const std::shared_ptr<abm::Graph>& graph) : graph_{graph} {};
+
+  //! Set id 
+  //! \param[in] agent_id Agent id assigned to the agent object
+  void set_id(graph::vertex_t agent_id) { this->agent_id_ = agent_id; }
+  //! Get id
+  graph::vertex_t get_id() const { return agent_id_; }
+
+  //! Set origin and initialize current node to origin
+  //! \param[in] origin Origin vertex assigned to agent
+  void set_origin(graph::vertex_t origin) { 
+    this->origin_ = origin;
+    this->current_node_ = origin;
+  }
+  //! Get origin
+  graph::vertex_t get_origin() const { return origin_; }
+
+  //! Set destination
+  //! \param[in] destination Destination vertex assigned to agent
+  void set_destination(graph::vertex_t destination) { this->destination_ = destination; }
+  //! Get destination
+  graph::vertex_t get_destination() const { return destination_; }
+
+  //! Set departure_time
+  //! \param[in] departure_time Departure time of an agent
+  void set_departure_time(graph::weight_t departure_time) { this->departure_time_ = departure_time; }
```

Comment:
Why are we using `graph::weight_t` to denote a double? Could you make this a double?

Reply:
Yes, but what I was thinking is, departure time and edge cost are both in the unit of time at the moment (edge cost can be in other units as well). Also, there is `using weight_t = double;`